app/models/katello/host/info_provider.rb

```diff
@@ -16,6 +16,15 @@ def host_info
         if host.content_facet.present?
           info['parameters']['kickstart_repository'] = host.content_facet.kickstart_repository.try(:label)
         end
+
+        if (rhsm_url = host.content_source&.rhsm_url)
```

Comment:
I learned that simply registering a content host directly to the server does not establish a content source and so these won't be set. Is that compatible with how you see this being used, or should they get the values from the default smart proxy at that point? Or is it a bug that content_source is nil in that scenario?

Reply:
I would consider that a bug but I'm not sure how others see it. IMHO every host that has content should have a `content_source` assigned. Otherwise you always need to remember to do `host.content_source || pulp3_primary`.

It also means that https://github.com/Katello/katello/pull/9780 may be based on an incorrect assumption and currently can't be used to reconfigure if the host is assigned to the primary.

Reply:
#9780 works OK. I can create an issue to track that unless you want to dive into it on this PR